QuantConnect.CoinbaseBrokerage.Tests/CoinbaseApiTests.cs

```diff
@@ -0,0 +1,214 @@
+ * limitations under the License.
+*/
+
+using System;
+using RestSharp;
+using System.Linq;
+using NUnit.Framework;
+using Newtonsoft.Json.Linq;
+using QuantConnect.Brokerages;
+using QuantConnect.Configuration;
+using System.Collections.Generic;
+using QuantConnect.CoinbaseBrokerage.Api;
+using QuantConnect.CoinbaseBrokerage.Models.Enums;
+using QuantConnect.CoinbaseBrokerage.Models.WebSocket;
+
+namespace QuantConnect.CoinbaseBrokerage.Tests
+{
+    [TestFixture]
+    [Explicit("Use tests for more clarification of API")]
+    public class CoinbaseApiTests
+    {
+        private CoinbaseApi CoinbaseApi { get; set; }
+
+        [SetUp]
+        public void Setup()
+        {
+            var apiKey = Config.Get("coinbase-api-key");
+            var apiKeySecret = Config.Get("coinbase-api-secret");
+
+            CoinbaseApi = CreateCoinbaseApi(apiKey, apiKeySecret);
+        }
+
+        [TestCase("", "")]
+        [TestCase("1", "2")]
+        public void InvalidAuthenticationCredentialsShouldThrowException(string apiKey, string apiKeySecret)
+        {
+            var coinbaseApi = CreateCoinbaseApi(apiKey, apiKeySecret);
+
+            // call random endpoint with incorrect credential
+            Assert.Throws<Exception>(() => coinbaseApi.GetAccounts());
+        }
+
+        [Test] 
+        public void GetListAccounts() 
+        {
+            var accounts = CoinbaseApi.GetAccounts();
+
+            Assert.Greater(accounts.Count(), 0);
+
+            foreach(var account in accounts)
+            {
+                Assert.IsTrue(account.Active);
+                Assert.IsNotEmpty(account.Name);
+                Assert.IsNotEmpty(account.Currency);
+                Assert.IsNotEmpty(account.AvailableBalance.Currency);
+                Assert.GreaterOrEqual(account.AvailableBalance.Value, 0);
+                Assert.That(account.CreatedAt, Is.LessThan(DateTime.UtcNow));
+            }
+        }
+
+        [Test]
+        public void GetListOrdersWithDifferentOrderStatus()
+        {
+            foreach (var orderStatus in (OrderStatus[])Enum.GetValues(typeof(OrderStatus)))
+            {
+                // Not supported order status request
+                if(OrderStatus.Pending == orderStatus)
+                {
+                    Assert.Throws<Exception>(() => CoinbaseApi.GetOrders(orderStatus));
+                    continue;
+                }
+
+                var orders = CoinbaseApi.GetOrders(orderStatus);
+                Assert.IsNotNull(orders);
+            }
+        }
+
+        [TestCase("BTC-USDC")]
+        public void GetProductCandlesWithDifferentCandleGranularity(string productId)
+        {
+            foreach (var candleGranularity in (CandleGranularity[])Enum.GetValues(typeof(CandleGranularity)))
+            {
+                var startDateTime = candleGranularity switch
+                {
+                    CandleGranularity.OneMinute => DateTime.UtcNow.AddMinutes(-10),
```

Comment:
Would be better to use a fixed reference date instead of Now for reproducibility

Reply:
Rewrote on hardcoded dateTime